src/server/pkg/worker/api_server.go

```diff
@@ -101,13 +113,21 @@ func (a *APIServer) getTaggedLogger(req *ProcessRequest) *taggedLogger {
 	result.template.JobID = req.JobID
 
 	// Add inputs' details to log metadata, so we can find these logs later
+	hash := sha256.New()
```

Comment:
Hash is used so that we can have an ID for each datum, this is used in logs and also as the path in the stats output so we don't have to do something like `_path_to_datum_a+path_to_datum_b`. I'll add a comment about this as well.